ocaml/xapi/nm.ml

```diff
@@ -642,13 +652,33 @@ let bring_pif_up ~__context ?(management_interface = false) (pif : API.ref_PIF)
               in
               Net.Interface.make_config dbg false interface_config) ;
           let new_ip =
-            match Net.Interface.get_ipv4_addr dbg bridge with
+            let addr =
+              match address_type with
+              | `IPv4 ->
+                  Net.Interface.get_ipv4_addr dbg bridge
+              | `IPv6 ->
+                  Net.Interface.get_ipv6_addr dbg bridge
+            in
+            match addr with
             | (ip, _) :: _ ->
                 Unix.string_of_inet_addr ip
             | [] ->
                 ""
           in
-          if new_ip <> rc.API.pIF_IP then (
+          let pif_ip =
+            match address_type with
+            | `IPv4 ->
+                rc.API.pIF_IP
+            | `IPv6 -> (
+              match rc.API.pIF_IPv6 with
+              | [] ->
+                  ""
+              | hd :: _ -> (
+                match String.split_on_char '/' hd with [ip; _] -> ip | _ -> ""
+              )
```

Comment:
Can you add a comment to show an example of what `pIF_IPv6` looks like? It's not obvious to me why it would contain slashes

And if it doesn't contain a slash, would it not be reasonable to assume the whole string is a valid IP? However, this should be obvious from an example

Reply:
Yes, the IPv6 are stored in the xapi db like this: `ipv6/netmask` instead of having a netmask field.
I'll add a comment.